_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,182 @@
+other direction like allowing Table API Pipelines being used in the DataStream API. The new Kafka
+source has been, as mentioned, exposed in the Table API and Flink SQL. In the Python API there have
+also been some improvements by e.g. supporting UDF chaining in the Python DataStream API and general
+improvements regarding the DataStream and Table API.
+
+## Unified Checkpointing Experience
+
+One of the biggest effort, that is not really user facing, but will have a huge impact is unifying
+the checkpointing experience between bounded and unbounded data streams. This his been a huge amount
+of work too. Essentially this means allowing checkpoints after some tasks are finished. This is
+often the case in heterogeneous environments mixing bounded and unbounded streams (and sources) and
+has a huge impact in the exactly once delivery guarantee.
+
+# Notable improvements regarding production readiness
+
+The previous section was all about improving the user experience, but Apache Flink became what it is
+by enabling use cases in stream processing that have not been possible before. Some of the biggest
+data processing use cases in the industry are built on top of Apache Flink.
+
+## Buffer debloating
+
+Running huge use cases processing a lot of data in Apache Flink means long checkpointing times and
+therefor a rather poor experience when it comes to fault tolerance. With unaligned checkpoints a
+feature to target this experience was recently added to Apache Flink. One of the reasons for the
+slow checkpoints are the huge size of them, this is mostly caused by data that is in the buffers. In
+this release the community will expose a beta feature called buffer debloating. This is essentially
+trying to minimise the data that are used before and after a task. They have been of fixed size
+until this release. Now they changed towards self-optimising themselves reducing the buffer size
+significantly and therefor reducing the checkpointing time a lot.
+
+## Fine grained resource and network buffer management
+
+Apache Flink is a data processor. Even by acknowledging stream and batch based use cases within
+those two types there are uncountable options on how a data processing use case might look. To get
+the optimum out of Flink there are now more options on how to manage resources moving from coarse
+grained resource management to a fine grained one.
+
+There is also an improvement on how to do network buffer management overcoming some existing
+limitations. All in all this should require less network memory.
+
+## Connector metrics
+
+Moving use cases into production usually also means increasing observability. Connectors are the
+entry and the exit for a Flink job and it usually makes sense to monitor what is happening there
+closely. The telemetry data of a connector can also provide important pointers to narrow down
+problems or bottlenecks. In 1.14. there are default metrics for connectors introduced and also
```

Comment:
I think we can score bonus points here my giving examples, like numRecordsIn for sources / numRecordsOut for sink, which _so_ many people asked for over the years.